### ISSUE TITLE
Add `string` constructor and `Dispose()` to `BSTR`

### ISSUE THREAD

**Issue by JeremyKuhne** [microsoft/CsWin32#718]
To ease interop it would be good to have a constructor that takes a `string` and implement `IDisposable` for freeing the `BSTR`. This is what we're doing in WinForms:

``` C#
public BSTR(string value) : this((char*)Marshal.StringToBSTR(value))
{
}

public void Dispose()
{
    Marshal.FreeBSTR((nint)Value);
    // Yes, I know this is unconventional, but we make the assertion that all invariant assertions aren't valid with disposed items.
    // Feel free to ignore this part of the suggestion if it violates sensibilities.
    fixed (char** c = &Value)
    {
        *c = null;
    }
}
```

**Comment by AArnott** [comment]
Regarding your 'unconventional part', I just wonder why you went to all the trouble to get a pointer to your own field, only to immediately dereference it. Wouldn't `Value = null` have been simpler?

**Comment by AArnott** [comment]
I had this implemented and ready to go. Then I thought about double-disposal. I gotta make sure that's safe, so I'd have to clear the Value field. 
Then I thought about how as a struct, this value will not be finalized and thus a native leak would occur if folks don't call Dispose properly. 
This might also set expectations that PWSTR and other types also offer string constructors, etc. I feel that we might be crossing a boundary where it's harder to know where the responsibility lies for allocating and freeing memory for the user. Since this doesn't add a *lot* of value (it doesn't really unblock anything), I'm going to Won't Fix for now.

Folks can always add this functionality themselves via a custom partial class if they choose.
